x-pack/plugins/features/common/feature_kibana_privileges.ts

```diff
@@ -14,6 +14,18 @@ export interface FeatureKibanaPrivileges {
    */
   excludeFromBasePrivileges?: boolean;
 
+  /**
+   * Whether or not this privilege should only be granted to `All Spaces *`. Should be used for features that do not
+   * support Spaces. Defaults to `false`.
+   */
+  requireAllSpaces?: boolean;
+
+  /**
+   * Whether or not this privilege should be hidden in the roles UI and disallowed on the API. Defaults to `false`.
+   * @deprecated
```

Comment:
Marking deprecated by default per @legrego's suggestion. Fleet may be the only one needing this for now and we know will support a read-only mode in the future. OTOH, we may be using `requireAllSpaces` for some time, so I'm not marking that one deprecated.